src/CodeCheckers/FunctionChecker.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace CodeCheckers;
+
+
+
+use Nette\Utils\Strings;
+
+class FunctionChecker
+{
+
+	public static function createChecker()
+	{
+		return function ($checker, $s) {
+			if (!$checker->is('php,phpt')) {
+				return;
+			}
+			if (Strings::match($s, '#^interface#m')) {
+				return;
+			}
+			if (!Strings::match($s, '#^class#m') && !Strings::match($s, '#^abstract class#m')) {
+				return;
+			}
+			foreach (explode("\n", $s) as $line) {
+
+				if ($matchAll = Strings::matchAll($line, '#^([\s])+function(.*)#m')) {
```

Comment:
odstranil jsem zbytečný `foreach (explode("\n", $s) as $line) {`
prochází to celý soubor najednou.